impeller/geometry/half.cc

```diff
@@ -0,0 +1,18 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "half.h"
+
+#include "flutter/fml/logging.h"
+
+namespace impeller {
+
+_Half ScalarToHalf(Scalar f) {
+#ifdef FML_OS_WIN
+  FML_LOG(ERROR) << "ScalarToHalf conversion on unsupported platform.";
```

Comment:
What action can a developer who sees this take? This seems like a bug in the engine rather than a message for the end developer. Maybe a DCHECK for testing purposes?

Reply:
For now there is no reason to support this on any platform besides macOS/iOS. I'm using a clang specific feature to (hopefully) get hardware supported conversions. I could add a software conversion too

Reply:
I could make this fatal?

Reply:
A DCHECK is fatal for our unit tests but not users. Making it fatal for users seems too severe to me, but I'm not sure I understand the implications of going down this code path on an unexpected platform.

Reply:
Using this code on an unsupported platform implied that you constructed a HalfVectorN type manually. Nowhere in impeller do we do this except for unit tests, as the only references to it should be in the generated headers. On an unsupported platform the headers will not contain HalfVectorN types, and thus the implicit constructor will never be invoked.

So the possibility of this happening would be similar to the case of leaving `exit(0)` in our code somewhere. I should make it FML_UNREACHABLE just to be sure though.